src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IBinaryOperatorExpression.cs

```diff
@@ -2865,5 +2865,1435 @@ void M(bool? a, P b, P c, bool? result)
+                    (NoConversion)
+                  Operand: 
+                    IBinaryOperation (BinaryOperatorKind.ConditionalOr) (OperationKind.BinaryOperator, Type: ?, IsInvalid) (Syntax: 'a || b')
+                      Left: 
+                        IParameterReferenceOperation: a (OperationKind.ParameterReference, Type: C, IsInvalid) (Syntax: 'a')
+                      Right: 
+                        IParameterReferenceOperation: b (OperationKind.ParameterReference, Type: C, IsInvalid) (Syntax: 'b')
+
+    Next (Regular) Block[B2]
+Block[B2] - Exit
+    Predecessors: [B1]
+    Statements (0)
+";
+            var expectedDiagnostics = new[] {
+                // file.cs(6,18): error CS0217: In order to be applicable as a short circuit operator a user-defined logical operator ('C.operator |(C?, C?)') must have the same return type and parameter types
+                //         result = a || b;
+                Diagnostic(ErrorCode.ERR_BadBoolOp, "a || b").WithArguments("C.operator |(C?, C?)").WithLocation(6, 18)
+            };
+
+            VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(source, expectedGraph, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [WorkItem(27044, "https://github.com/dotnet/roslyn/issues/27044")]
+        [Fact]
+        public void LogicalUserDefinedFlow_14()
+        {
+            string source = @"
+struct C
+{
+    void M(C a, C b, C result)
+/*<bind>*/{
+        result = a || b;
+    }/*</bind>*/
+
+    public static C operator &(C x, C y) => throw null;
+    public static C operator |(C x, C y) => throw null;
+    public static bool operator true(C? c) => throw null;
+    public static bool operator false(C? c) => throw null;
+}
+";
+            // Even though there is no error, compiler emits an invalid IL (https://github.com/dotnet/roslyn/issues/27044)
+            // Treating this case as an error case for now.
+            string expectedGraph = @"
+Block[B0] - Entry
+    Statements (0)
+    Next (Regular) Block[B1]
+Block[B1] - Block
+    Predecessors: [B0]
+    Statements (2)
+        IFlowCaptureOperation: 0 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'result')
+          Value: 
+            IParameterReferenceOperation: result (OperationKind.ParameterReference, Type: C) (Syntax: 'result')
+
+        IFlowCaptureOperation: 1 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'a')
+          Value: 
+            IParameterReferenceOperation: a (OperationKind.ParameterReference, Type: C) (Syntax: 'a')
+
+    Jump if False (Regular) to Block[B3]
+        IUnaryOperation (UnaryOperatorKind.True) (OperatorMethod: System.Boolean C.op_True(C? c)) (OperationKind.UnaryOperator, Type: System.Boolean, IsImplicit) (Syntax: 'a')
+          Operand: 
+            IInvalidOperation (OperationKind.Invalid, Type: C?, IsImplicit) (Syntax: 'a')
```

Comment:
Do we know what bound node this corresponds to? Seems weird that there's an IInvalidOperation in compiling C# code.

Reply:
See comment above

---
In reply to: [190711693](https://github.com/dotnet/roslyn/pull/27070#discussion_r190711693) [](ancestors = 190711693)

Reply:
Ah, totally missed the comment, thanks.

---
In reply to: [190712265](https://github.com/dotnet/roslyn/pull/27070#discussion_r190712265) [](ancestors = 190712265,190711693)